src/details/ArborX_DetailsDistributor.hpp

```diff
@@ -62,6 +62,91 @@ struct Result
   }
 };
 
+// Assuming that batched_ranks might contain elements multiply, but duplicates
+// only are not separated by other elements, compute the array of indices that
+// sort the input array (in reverse order) but also return the sorted unique
+// elements in that array with the corresponding element counts and displacement
+// (offsets).
+//  This overload uses batched ranks and offsets.
+template <typename InputView, typename OutputView>
+static void
+DetermineBufferLayout(InputView batched_ranks, InputView batched_offsets,
+                      OutputView permutation_indices,
+                      std::vector<int> &unique_ranks, std::vector<int> &counts,
+                      std::vector<int> &offsets)
+{
+  ARBORX_ASSERT(unique_ranks.empty());
+  ARBORX_ASSERT(offsets.empty());
+  ARBORX_ASSERT(counts.empty());
+  ARBORX_ASSERT(permutation_indices.extent_int(0) ==
+                lastElement(batched_offsets));
+  ARBORX_ASSERT(batched_ranks.size() + 1 == batched_offsets.size());
+  static_assert(
+      std::is_same<typename InputView::non_const_value_type, int>::value, "");
+  static_assert(std::is_same<typename OutputView::value_type, int>::value, "");
+
+  offsets.push_back(0);
+
+  auto const n = batched_ranks.size();
+  if (n == 0 || lastElement(batched_offsets) == 0)
+    return;
```

Comment:
If we `n != 0` but we return early, should not we have at least one unique rank? I think it would still be reasonable to initialize `unique_ranks`, `count` and `offsets`, and maybe just make `permutation_indices` empty or something.

Reply:
This implementation is matching the previous behavior when expanding the batches and calling the non-batched `sortAndDetermineBufferLayout`.